### PR TITLE
Improve inst seg perf

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- Ticket no. 128949
- Stack images as possible for all tasks
- Pad images during instance segmentation data pipeline as possible
- Add non_blocking option to instance segmentation data preprocessors

### How to test
Median train iter time for this command
```console
otx train --data_root /home/vinnamki/dataset/instance_seg/coco_car_person_medium --config src/otx/recipe/instance_segmentation/maskrcnn_efficientnetb2b.yaml 
```
is improved,
| Before | After |
| :-: | :-: |
| 0.247 | 0.236 |

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

File: src/otx/cli/__init__.py [hunk 1]
```diff
@@ -11,7 +13,10 @@ def main() -> None:
 
     This function is a single entry point for all OTX CLI related operations:
     """
+    start = time()
```
Comment (sovrasov) [inline src/otx/cli/__init__.py:16]: Is that  change intended?

File: src/otx/recipe/instance_segmentation/maskrcnn_swint.yaml [hunk 2]
```diff
@@ -85,6 +89,8 @@ overrides:
             scale:
               - 1344
               - 1344
+          - type: Pad
```
Comment (sovrasov) [inline src/otx/recipe/instance_segmentation/maskrcnn_swint.yaml:92]: Are the images padded directly to the target resolution (here 1344x1344)?

File: src/otx/cli/__init__.py (hunk 1 above)
Comment (vinnamkim) [reply src/otx/cli/__init__.py:16]: No but is there any problem with it? @harimkang

File: src/otx/recipe/instance_segmentation/maskrcnn_swint.yaml (hunk 2 above)
Comment (vinnamkim) [reply src/otx/recipe/instance_segmentation/maskrcnn_swint.yaml:92]: No, they are padded to the closest integer to their width or height but divisible with the number 32.
https://github.com/openvinotoolkit/training_extensions/blob/66f9d304dfca73525a76728ff7ad34ae37a80954/src/otx/recipe/instance_segmentation/maskrcnn_swint.yaml#L93

File: src/otx/cli/__init__.py (hunk 1 above)
Comment (harimkang) [reply src/otx/cli/__init__.py:16]: I don't see a problem with that.

**Review by eugene123tw — APPROVED** [review]

**Pull request merged**